ZombieSharp/Statistic.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Dapper;
+using Microsoft.Extensions.Logging;
+using MySql.Data.MySqlClient;
+using Newtonsoft.Json;
+
+namespace ZombieSharp
+{
+    public partial class ZombieSharp
+    {
+        private StatsDatabase statsConfig;
+        MySqlConnection connection = null;
+
+        private void StatsOnLoad()
+        {
+            var statsConfigPath = Path.Combine(ModuleDirectory, $"../../configs/zombiesharp/database.json");
+            statsConfig = JsonConvert.DeserializeObject<StatsDatabase>(File.ReadAllText(statsConfigPath));
+            connection = StatsConnect();
+        }
+
+        private MySqlConnection StatsConnect()
+        {
+            var query = $"Server={statsConfig.hostname}; database={statsConfig.database}; UID={statsConfig.username}; password={statsConfig.password}";
+            var conn = new MySqlConnection(query);
+
+            if(conn == null)
+            {
+                Logger.LogError("[Stats] Connection is null.");
+                return null;
+            }
+
+            conn.Open();
+
+            return conn;
+        }
+
+        private async Task StatsSetData(CCSPlayerController client, int damage = 0, int kill = 0, int infect = 0)
+        {
+            if (client == null)
+            {
+                Logger.LogError("[Stats] client is null.");
+                return;
+            }
+
+            if (connection == null)
+            {
+                return;
+            }
+
+            var clientData = GetClientStatsData(client);
+
+            var query = $"INSERT INTO `zsharp_stats` (player_name, steam_auth, total_dmg, total_kill, total_infect, last_join) " +
+                    $"VALUES (\"{client.PlayerName}\", \"{client.AuthorizedSteamID.SteamId3}\", {damage}, {kill}, {infect}, \"{DateTime.Now.ToString("yyyy'-'MM'-'dd")}\") ";
+
+            if(clientData != null)
+            {
+                query += $"ON DUPLICATE KEY UPDATE player_name = \"{client.PlayerName}\", total_dmg = {damage + clientData.TotalDamage}, total_kill = {kill + clientData.TotalKill}, total_infect = {infect + clientData.TotalKill}, last_join = \"{DateTime.Now.ToString("yyyy'-'MM'-'dd")}\"";
+            }
+
+            // Logger.LogInformation($"[Stats] Query: {query};");
+
+            await connection.ExecuteAsync($"{query};");
+        }
+
+        private ClientStatsData GetClientStatsData(CCSPlayerController client)
+        {
+            if (client == null)
+                return null;
+
+            var conn = StatsConnect();
+
+            if (connection == null)
+                return null;
+
+            var data = new MySqlCommand($"SELECT * FROM `zsharp_stats` WHERE steam_auth = \"{client.AuthorizedSteamID.SteamId3}\";", conn);
```

Comment:
similar to previous. Avoid string interpolations

Reply:
use `using` keyword for `IDisposable` to automatically control your resources: `using var command = new MySqlCommand( ... );`

now if any exception throwed, your code will handle for it and dispose resources